libraries/extensions/dora-record/src/main.rs

```diff
@@ -0,0 +1,134 @@
+use chrono::{DateTime, Utc};
+use dora_node_api::{
+    self,
+    arrow::{
+        array::{make_array, Array, Int64Array, ListArray, StringArray},
+        buffer::{OffsetBuffer, ScalarBuffer},
+        datatypes::{DataType, Field, Schema},
+        ipc::writer::FileWriter,
+        record_batch::RecordBatch,
+    },
+    DoraNode, Event, Metadata,
+};
+use dora_tracing::telemetry::deserialize_to_hashmap;
+use eyre::{Context, ContextCompat};
+use std::{collections::HashMap, fs::File, sync::Arc};
+
+// Remove once arrow-rs 48.0 is published with access to writer schema.
+// See: https://github.com/apache/arrow-rs/pull/4940
+struct WriterContext {
+    writer: FileWriter<File>,
+    schema: Arc<Schema>,
+}
```

Comment:
PR #353 updated `arrow` to v48.0.0 in https://github.com/dora-rs/dora/pull/353/commits/8cc5837cfa38995e3829213ae8d40c9145910e96.

Reply:
ah nice!